include/wasmtime.hh

```diff
@@ -35,16 +35,64 @@
 #include <any>
 #include <initializer_list>
 #include <iostream>
+#include <limits>
 #include <memory>
 #include <optional>
-#include <span>
 #include <variant>
 #include <vector>
+#if __cplusplus > 201703L && __has_include(<span>)
+#include <span>
+#endif
 
 #include "wasmtime.h"
 
 namespace wasmtime {
 
+/**
+ * \brief Span class used when c++20 is not available
+ * @tparam T Type of data
+ * @tparam Extent Static size of data refered by Span class
+ */
+template <typename T,
+          std::size_t Extent = std::numeric_limits<std::size_t>::max()>
+class Span {
+public:
+  /// \brief Type used to iterate over this span (a raw pointer)
+  using iterator = T *;
+
+  /// \brief Constructor of Span class
+  Span(T *t, std::size_t n) : ptr_{t}, size_{n} {}
+
+  /// \brief Returns item by index
+  T &operator[](ptrdiff_t idx) const { return ptr_[idx]; }
+
+  /// \brief Returns pointer to data
+  T *data() const { return ptr_; }
+
+  /// \brief Returns number of data that referred by Span class
+  std::size_t size() const { return size_; }
+
+  /// \brief Returns begin iterator
+  iterator begin() const { return ptr_; }
+
+  /// \brief Returns end iterator
+  iterator end() const { return ptr_ + size_; }
+
+  /// \brief Returns size in bytes
+  std::size_t size_bytes() const { return sizeof(T) * size_; }
+
+#if __cplusplus > 201703L && __has_include(<span>)
```

Comment:
I was seeing that `__has_include(<span>)` could return true for C++17 but the `span` type wasn't available then I thought? Maybe it was my compiler? In any case though I found the feature test of `__cpp_lib_span` to work at this point

Reply:
Updated